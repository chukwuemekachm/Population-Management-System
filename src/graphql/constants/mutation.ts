import { gql } from 'apollo-server-express';

export const SIGNUP_MUTATION = gql`
  mutation signup($user: SignupInput) {
    signup(user: $user) {
      user {
        email
        firstName
        lastName
      }
    }
  }
`;
