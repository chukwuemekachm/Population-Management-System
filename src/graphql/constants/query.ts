import { gql } from 'apollo-server-express';

export const GET_LOCATIONS_QUERY = gql`
  query getLocations($filter: LocationFilterInput) {
    getLocations(filter: $filter) {
      locationName
    }
  }
`;
