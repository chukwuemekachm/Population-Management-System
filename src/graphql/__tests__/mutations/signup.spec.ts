import { createTestClient } from 'apollo-server-testing';

import graphqlTestServer from '../../__mocks__/server.graphql';
import prismaTestMockClient from '../../__mocks__/prisma';
import { SIGNUP_MUTATION } from '../../constants';
import { userInputs } from '../../fixtures/userFixtures';

const { mutate } = createTestClient(graphqlTestServer);

describe('signup mutation', () => {
  beforeAll(async () => {
    await prismaTestMockClient.deleteManyUsers();
    await prismaTestMockClient.createUser(userInputs[1]);
  });
  afterAll(async () => {
    await prismaTestMockClient.deleteManyUsers();
  });

  test('should return created user profile', async () => {
    const payload = {
      variables: {
        user: userInputs[2],
      },
    };
    const { errors, data } = await mutate({
      mutation: SIGNUP_MUTATION,
      ...payload,
    });
    expect(errors).toBeUndefined();
    expect(data).toBeDefined();
    expect(data).toMatchSnapshot();
  });

  test('should return validation errors', async () => {
    const payload = {
      variables: {
        user: userInputs[0],
      },
    };
    const { errors, data } = await mutate({
      mutation: SIGNUP_MUTATION,
      ...payload,
    });
    expect(errors).toBeDefined();
    expect(data).toBeNull();
    expect(errors).toMatchSnapshot();
  });

  test('should return a user exist error', async () => {
    const payload = {
      variables: {
        user: userInputs[2],
      },
    };
    const { errors, data } = await mutate({
      mutation: SIGNUP_MUTATION,
      ...payload,
    });
    expect(errors).toBeDefined();
    expect(data).toBeNull();
    expect(errors).toMatchSnapshot();
  });
});
