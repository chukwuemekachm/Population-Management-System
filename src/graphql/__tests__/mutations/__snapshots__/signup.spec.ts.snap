// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup mutation should return a user exist error 1`] = `
Array [
  [GraphQLError: User with email exists],
]
`;

exports[`signup mutation should return created user profile 1`] = `
Object {
  "signup": Object {
    "user": Object {
      "email": "test2@mail.com",
      "firstName": "Test2",
      "lastName": "User2",
    },
  },
}
`;

exports[`signup mutation should return validation errors 1`] = `
Array [
  [GraphQLError: Some field(s) are failing validation],
]
`;
